modal/_output.py

```diff
@@ -355,12 +356,15 @@ async def _get_logs():
                             else:
                                 # If we're not showing progress, there's no need to buffer lines,
                                 # because the progress spinner can't interfere with output.
-                                #
-                                # Flush data in chunks because write is blocking.
-                                # TODO: make non-blocking write work.
-                                for i in range(0, len(log.data), 64):
-                                    self.stdout.write(log.data[i : i + 64])
-                                    self.stdout.flush()
+
+                                written = 0
+                                while written < len(log.data):
```

Comment:
It probably doesn't matter in this case since it either writes all or errors in each loop iteration, but this should probably compare to the length of the utf8-encoded log.data to be correct.

Reply:
Good point! Actually the third case is also possible: `os.write` does sometimes write part of the data you asked it to and return how many bytes it wrote. It's going to be quite tricky to trigger that intentionally, but it's a bug nonetheless. Fixing here: https://github.com/modal-labs/modal-client/pull/195